app/models/user.rb

```diff
@@ -284,11 +284,9 @@ def may_view_solution?(solution)
     solution.viewable_by?(self)
   end
 
-  memoize
+  # Don't memoize this as it can change when someone uploads their photo
   def avatar_url
-    return Rails.application.routes.url_helpers.url_for(avatar.variant(:thumb)) if avatar.attached?
-
-    super.presence || "#{Exercism.config.website_icons_host}/placeholders/user-avatar.svg"
+    "#{Exercism.config.website_avatars_host}/avatars/#{id}/#{version}"
```

Comment:
This is the key change that means rather than looking up the url or file ourselves, we always just push someone to `assets.exercism.org`, which either renders the cached version or hits the new Avatars controller above (but obviously as a seperate request)